src/components/Settings.jsx

```diff
@@ -41,22 +44,39 @@ class Settings extends Component {
     this.getUserData();
   }
 
+  toggleSettingsView = () => {
+    let { settingsHeader, settingsBlock } = this.state;
+    settingsHeader = settingsHeader === 'Hide Settings' ? 'Show Settings' : 'Hide Settings';
+    settingsBlock = settingsBlock === 'showBlock' ? 'hideBlock' : 'showBlock';
+    this.setState({ settingsHeader, settingsBlock });
+  }
+
   render() {
     return (
       <div>
         <div>
-          <h3>Settings</h3>
-          <Details
-            saveUserData={this.saveUserData}
-            email={this.state.email}
-            petName={this.state.petName}
-            roomName={this.state.roomName}
-          />
-          <Whitelist
-            saveUserData={this.saveUserData}
-            email={this.state.email}
-            phoneNumbers={this.state.phoneNumbers}
-          />
+          <h3>
+            <div
+              className="settingsToggle"
+              onClick={this.toggleSettingsView}
+              role="button"
+            >
+              {this.state.settingsHeader}
+            </div>
+          </h3>
+          <div className={this.state.settingsBlock}>
```

Comment:
glad you found out you could do this. but what if I told you you didn't need to use classes at all

Reply:
you could have a Boolean state right here {this.state.isShowing && <div>settings/whitelist</div>}